src/compile/mark/bar.ts

```diff
@@ -17,6 +17,10 @@ export namespace bar {
 
     const stack = model.stack();
```

Comment:
Ah I see. Currently, it doesn't affect the truthy of `stack` when `x2` or `y2` is specified. It essentially ignores it. Although, the scales are made with a union of the two domains(`x` + `x2`). I am not sure how we should handle that. Maybe create the stack before scales and then check for the `stack` when creating `domain` for the scales. 

Also, I wanted to ask if we wanna allow to make stacked bar charts with `x2` instead of `x`. That might require a little work but if not then it seems to be okay.